src/platform/plugins/shared/chart_expressions/expression_xy/public/components/xy_chart.tsx

```diff
@@ -873,7 +879,7 @@ export function XYChart({
               onBrushEnd={interactive ? (brushHandler as BrushEndListener) : undefined}
               onElementClick={interactive ? clickHandler : undefined}
               legendAction={
-                interactive
+                interactive && canCreateFilters
```

Comment:
You can find another usage of the legendAction in the partition chart https://github.com/markov00/kibana/blob/83ee39b9d048a591dd837b41638d5c7f952db9b9/src/platform/plugins/shared/chart_expressions/expression_partition_vis/public/components/partition_vis_component.tsx#L339

Reply:
@markov00 they don't have actions in ES|QL charts, this is why I didnt make any changes there. Am I missing something?

<img width="377" alt="image" src="https://github.com/user-attachments/assets/36ccf44c-1221-493f-9440-015f1787bf8a" />

Reply:
So you asked for:
>I didnt find other charts with legend actions but let me know if I forgot anything

Partition has legend actions. It seems completely disabled if it is an ESQL chart, but this differs from what you just did here in this PR, where the legend actions are enabled in discover.

There is also a collateral issue here: you are allowing filtering from the legend and/or by clicking on the chart, but the tooltipActions are disabled (right click and press filter). They should work at the same way.

Reply:
> I didnt find other charts with legend actions but let me know if I forgot anything

If they were legend actions that were working for ES|QL. This PR is about it, not adding extra functionality 🤔